domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -170,17 +175,16 @@ public async Task<ActionResult<EditTransferAgreementEndDate>> EditEndDate(Guid i
     }
 
     [ProducesResponseType(typeof(string), 200)]
+    [ProducesResponseType(typeof(string), 500)]
     [HttpPost("wallet-deposit-endpoint")]
     public async Task<ActionResult> CreateWalletDepositEndpoint()
     {
-        var issuer = User.FindIssuerClaim();
-        var audience = User.FindAudienceClaim();
-        var subject = User.FindSubjectGuidClaim();
-        var name = User.FindSubjectNameClaim();
-
-        var base64String = await walletDepositEndpointService.CreateWalletDepositWithToken(new JwtToken(issuer, audience, subject, name));
-
-        return Ok(new { result = base64String });
+        if (AuthenticationHeaderValue.TryParse(httpContextAccessor.HttpContext?.Request.Headers["Authorization"], out var authentication))
+        {
+            var base64String = await walletDepositEndpointService.CreateWalletDepositWithToken(authentication.Parameter);
+            return Ok(new { result = base64String });
+        }
+        return StatusCode(500);
```

Comment:
It seems wrong to send a status code 500 due to the fact that there is no jwt-token. Additionally, the controller requires all requests to be authenticated, so there will always be a jwt-token.
I suggest to not use the TryParse-method and instead use the Parse-method: https://learn.microsoft.com/en-us/dotnet/api/system.net.http.headers.authenticationheadervalue.parse?view=net-7.0#system-net-http-headers-authenticationheadervalue-parse(system-string). With this approach there is no branching to "what if we cannot get the jwt token?". Should this fail in the future, it will throw an exception with a saying message and the API client will still get a 500 Status code.

Reply:
I have switched to parse, assuming the jwt is always correctly formatted. if not it will throw an error.